### ISSUE TITLE
[Implementation] Function that stops traversal after a certain condition has been met

### ISSUE THREAD

**Issue by valeriehan6** [cmsc22000-project-2018/redis-tries#30]
Requested by spellcheck team. 

`relatives_t* related_strings(int max, int (*distance)(char* str1, char* str2), char* curr, trie_t* t);`

Takes in:
- int max
- function that takes in two char* and compares their distances and returns an int
- string that's being compared
- trie_t *t that's being searched for strings with distance at most max

Returns:
- Allocated linked list with string and corresponding distance

**Comment by valeriehan6** [comment]
commit e6fa080bec13733c9bd62c17569ebad256c4e515 - Currently waiting on spellcheck team to send me the distance function

**Comment by valeriehan6** [comment]
Closing because the spellcheck team no longer needs this function. Leaving the medium label because of the amount of work I had already done on the issue.
